test/test_weaviate.py

```diff
@@ -0,0 +1,313 @@
+
+def test_get_all_documents_with_incorrect_filter_name(document_store_with_docs):
+    documents = document_store_with_docs.get_all_documents(filters={"incorrectmetafield": ["test2"]})
+    assert len(documents) == 0
+
+
+def test_get_all_documents_with_incorrect_filter_value(document_store_with_docs):
+    documents = document_store_with_docs.get_all_documents(filters={"metafield": ["incorrect_value"]})
+    assert len(documents) == 0
+
+def test_get_documents_by_id(document_store_with_docs):
+    documents = document_store_with_docs.get_all_documents()
+    doc = document_store_with_docs.get_document_by_id(documents[0].id)
+    assert doc.id == documents[0].id
+    assert doc.text == documents[0].text
+
+@pytest.mark.parametrize("document_store", ["weaviate"], indirect=True)
+def test_get_document_count(document_store):
+    document_store.write_documents(DOCUMENTS)
+    assert document_store.get_document_count() == 5
+    assert document_store.get_document_count(filters={"key": ["a"]}) == 1
+    assert document_store.get_document_count(filters={"key": ["b"]}) == 4
+
+@pytest.mark.parametrize("document_store", ["weaviate"], indirect=True)
+@pytest.mark.parametrize("batch_size", [2])
+def test_weaviate_write_docs(document_store, batch_size):
+    # Write in small batches
+    for i in range(0, len(DOCUMENTS), batch_size):
+        document_store.write_documents(DOCUMENTS[i: i + batch_size])
+
+    documents_indexed = document_store.get_all_documents()
+    assert len(documents_indexed) == len(DOCUMENTS)
+
+    documents_indexed = document_store.get_all_documents(batch_size=batch_size)
+    assert len(documents_indexed) == len(DOCUMENTS)
+
+@pytest.mark.parametrize("document_store", ["weaviate"], indirect=True)
+def test_get_all_document_filter_duplicate_value(document_store):
+    documents = [
+        Document(
+            text="Doc1",
+            meta={"fone": "f0"},
+            id = get_uuid(),
+            embedding= np.random.rand(embedding_dim).astype(np.float32)
+        ),
+        Document(
+            text="Doc1",
+            meta={"fone": "f1", "metaid": "0"},
+            id = get_uuid(),
+            embedding = np.random.rand(embedding_dim).astype(np.float32)
+        ),
+        Document(
+            text="Doc2",
+            meta={"fthree": "f0"},
+            id = get_uuid(),
+            embedding=np.random.rand(embedding_dim).astype(np.float32)
+        )
+    ]
+    document_store.write_documents(documents)
+    documents = document_store.get_all_documents(filters={"fone": ["f1"]})
+    assert documents[0].text == "Doc1"
+    assert len(documents) == 1
+    assert {d.meta["metaid"] for d in documents} == {"0"}
+
+@pytest.mark.parametrize("document_store", ["weaviate"], indirect=True)
+def test_get_all_documents_generator(document_store):
+    document_store.write_documents(DOCUMENTS)
+    assert len(list(document_store.get_all_documents_generator(batch_size=2))) == 5
+
+@pytest.mark.elasticsearch
```

Comment:
This elasticsearch marker seems a left-over from copy pasting. 

However, I would suggest that we add `@pytest.mark.weaviate` to all tests in here and allow to select/deselect them when running pytest. We have this already for other test groups (e.g. elasticsearch) and you can see how it's implemented [here](https://github.com/deepset-ai/haystack/blob/master/test/conftest.py#L48) and how it can be used [here](https://github.com/deepset-ai/haystack#heart-contributing)

Reply:
my bad! This is fixed now.